sql/driver/wire.go

```diff
@@ -80,14 +80,15 @@ func (d Datum) Value() (driver.Value, error) {
 		val = t.BytesVal
 	case *Datum_StringVal:
 		val = t.StringVal
+	case *Datum_DateVal:
+		val = t.DateVal.GoUTCTime()
 	case *Datum_TimeVal:
-		val = t.TimeVal.GoTime().UTC()
+		val = t.TimeVal.GoUTCTime()
+	case *Datum_IntervalVal:
+		val = time.Duration(t.IntervalVal)
 	}
 
-	if driver.IsValue(val) {
-		return val, nil
-	}
-	return nil, util.Errorf("unsupported type %T", val)
```

Comment:
Ah, I see. It's the `else` for the `IsValue` check; I was reading it as the `default` for the type switch. Speaking of which, we should probably have one.